lib/text/vtt_text_parser.js

```diff
@@ -219,6 +222,75 @@ shaka.text.VttTextParser = class {
     return cue;
   }
 
+  /**
+   * Parses a WebVTT styles from the given payload.
+   *
+   * @param {string} payload
+   * @param {!shaka.text.Cue} rootCue
+   */
+  static parseCueStyles(payload, rootCue) {
+    const xmlPayload = '<span>' + payload + '</span>';
+    const element = shaka.util.XmlUtils.parseXmlString(xmlPayload, 'span');
+    if (element) {
+      /** @type {Array.<!shaka.extern.Cue>} */
+      const cues = [];
+      const VttTextParser = shaka.text.VttTextParser;
+      const childNodes = [...element.childNodes];
+      if (childNodes.length == 1) {
+        const childNode = childNodes[0];
+        if (childNode.nodeType == Node.TEXT_NODE ||
+            childNode.nodeType == Node.CDATA_SECTION_NODE) {
+          rootCue.payload = payload;
+          return;
+        }
+      }
+      for (const childElement of childNodes) {
+        VttTextParser.generateCueFromElement_(childElement, rootCue, cues);
+      }
+      rootCue.nestedCues = cues;
+    } else {
+      rootCue.payload = payload;
+    }
+  }
+
+  /**
+   * @param {!Node} element
+   * @param {Array.<!shaka.extern.Cue>} cues
+   * @private
+   */
+  static generateCueFromElement_(element, rootCue, cues) {
+    const nestedCue = rootCue.clone();
+    if (element.nodeType === Node.ELEMENT_NODE && element.nodeName) {
+      const bold = shaka.text.Cue.fontWeight.BOLD;
+      const italic = shaka.text.Cue.fontStyle.ITALIC;
+      const underline = shaka.text.Cue.textDecoration.UNDERLINE;
+      const tags = element.nodeName.split(/[ .]+/).filter(Boolean);
+      for (const tag of tags) {
+        switch (tag) {
+          case 'b':
+            nestedCue.fontWeight = bold;
+            break;
+          case 'i':
+            nestedCue.fontStyle = italic;
+            break;
+          case 'u':
+            nestedCue.textDecoration.push(underline);
+            break;
+        }
+      }
+    }
+    const contents = shaka.util.XmlUtils.getContents(element);
+    if (contents) {
+      nestedCue.payload = contents;
+      cues.push(nestedCue);
+    } else {
```

Comment:
It might be worth a comment inside the `else` block that this branch is taken when the element contains non-text nodes.  I had to read the implementation of `XmlUtils.getContents` to find that out.

Reply:
Ok, I'm going to make it better understood.